File: 69.py
#69. Sqrt(x)

class Solution(object):
    def mySqrt(self, x):
        if x <= 1:
            return x

        i = 1
        sq = i * i
        while x >= sq:
            i += 1
            sq = i * i
        
        return i - 1

File: 69.java
#69. Sqrt(x)

class Solution {
    public int mySqrt(int x) {
        if (x <= 1) {
            return x;
        }

        int i = 1;
        while (i <= x / i) { // Avoids overflow by using division
            i++;
        }
        return i - 1;
    }
}
